internal/services/communication/email_service_domain_sender_username_resource.go

```diff
@@ -0,0 +1,225 @@
+			ValidateFunc: validation.StringLenBetween(1, 253),
+		},
+	}
+}
+
+func (EmailCommunicationServiceDomainSenderUsernameResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (EmailCommunicationServiceDomainSenderUsernameResource) ModelObject() interface{} {
+	return &EmailCommunicationServiceDomainSenderUsernameModel{}
+}
+
+func (EmailCommunicationServiceDomainSenderUsernameResource) ResourceType() string {
+	return "azurerm_email_communication_service_domain_sender_username"
+}
+
+func (r EmailCommunicationServiceDomainSenderUsernameResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			client := metadata.Client.Communication.SenderUsernamesClient
+
+			var model EmailCommunicationServiceDomainSenderUsernameModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			eMailServiceDomainID, err := senderusernames.ParseDomainID(model.EMailServiceDomainID)
+			if err != nil {
+				return fmt.Errorf("parsing parent email_service_domain_id: %+v", err)
+			}
+
+			id := senderusernames.NewSenderUsernameID(subscriptionId, eMailServiceDomainID.ResourceGroupName, eMailServiceDomainID.EmailServiceName, eMailServiceDomainID.DomainName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := senderusernames.SenderUsernameResource{
+				Properties: &senderusernames.SenderUsernameProperties{
+					Username: model.Username,
+				},
+			}
+
+			if v := model.DisplayName; v != "" {
+				parameters.Properties.DisplayName = pointer.To(v)
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, parameters); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r EmailCommunicationServiceDomainSenderUsernameResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Communication.SenderUsernamesClient
+
+			var model EmailCommunicationServiceDomainSenderUsernameModel
+
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			id, err := senderusernames.ParseSenderUsernameID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil || existing.Model == nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			senderUsername := *existing.Model
+
+			props := pointer.From(senderUsername.Properties)
```

Comment:
could you compare `model` and `properties` with `nil` separately to keep consistent with the [contributing doc](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md#step-3-scaffold-an-emptynew-resource)?

Reply:
Updated